src/simplewallet/simplewallet.cpp

```diff
@@ -2209,19 +2295,26 @@ bool simple_wallet::transfer_main(int transfer_type, const std::vector<std::stri
     local_args.pop_back();
   }
 
+  bool is_subaddress = false;
   vector<cryptonote::tx_destination_entry> dsts;
   for (size_t i = 0; i < local_args.size(); i += 2)
   {
+    cryptonote::address_parse_info info;
     cryptonote::tx_destination_entry de;
-    bool has_payment_id;
-    crypto::hash8 new_payment_id;
-    if (!cryptonote::get_account_address_from_str_or_url(de.addr, has_payment_id, new_payment_id, m_wallet->testnet(), local_args[i]))
+    if (!cryptonote::get_account_address_from_str_or_url(info, m_wallet->testnet(), local_args[i]))
     {
       fail_msg_writer() << tr("failed to parse address");
       return true;
     }
+    is_subaddress = info.is_subaddress;
+    if (info.is_subaddress && local_args.size() > 2)
+    {
+      fail_msg_writer() << tr("Subaddresses (those starting with '5') cannot be used for multi-destination transfer");
```

Comment:
The subaddress prefix is chosen such that the address string always starts with '5' and 'B' for the mainnet and the testnet, respectively.

Here's how the first character of the base58 encoding is determined:
 - The encoder sequentially processes the data stream 8 bytes at a time and converts the block into a string of 11 base58 characters (since `58^10 < 2^64 < 58^11`).

- The prefix (18, 19, 26, 27 for the mainnet, or 53, 54, 63, 64 for the testnet) is stored in the first 1 byte, and the remaining 7 bytes are set by the arbitrary address public keys. This 8-byte data is stored in a 64bit number `uint64_t num` with big endian which is in the range of `(prefix)*2^56 <= num <= prefix*2^56 + 2^56 - 1`.

- The `i`-th character of the 11-char string is determined by `alphabet[pos]` where `pos = (num / 58^(10-i)) % 58`. Specifically, the first character is determined by `pos = (num / 58^10) % 58`.

```
alphabet = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";

18 * 2^56 / 58^10 = 3.01073358131 --> '4'
19 * 2^56 / 58^10 = 3.17799655805 --> '4'
20 * 2^56 / 58^10 = 3.34525953479 --> '4'

26 * 2^56 / 58^10 = 4.34883739523 --> '5'
27 * 2^56 / 58^10 = 4.51610037197 --> '5'
28 * 2^56 / 58^10 = 4.68336334871 --> '5'

53 * 2^56 / 58^10 = 8.8649377672 --> '9'
54 * 2^56 / 58^10 = 9.03220074394 --> 'A'
55 * 2^56 / 58^10 = 9.19946372068 --> 'A'

63 * 2^56 / 58^10 = 10.5375675346 --> 'B'
64 * 2^56 / 58^10 = 10.7048305113 --> 'B'
65 * 2^56 / 58^10 = 10.8720934881 --> 'B'
```